crates/holochain_keystore/src/meta_lair_client.rs

```diff
@@ -141,6 +141,87 @@ impl MetaLairClient {
         Ok(MetaLairClient(inner, c_check_send))
     }
 
+    /// Create a MetaLairClient from a LairClient
+    pub async fn from_client(client: LairClient) -> LairResult<Self> {
+        let inner = Arc::new(Mutex::new(client));
+
+        let (c_check_send, mut c_check_recv) = tokio::sync::mpsc::unbounded_channel();
```

Comment:
All this connection checking code should not be duplicated. Please refactor so it is shared.

Reply:
It's not possible to share it entirely because it differs [here](https://github.com/holochain/holochain/blob/e8d651e50c71e5c400a971da07bc5715c048b583/crates/holochain_keystore/src/meta_lair_client.rs#L115-L129)

Reply:
hang on, that's the bit doing the reconnect. Your new logic doesn't need the reconnect loop at all, because there isn't anything to reconnect to, no? Can you just put in a stub sender that doesn't go anywhere and delete all the rest?

Reply:
To be honest, I still don't fully understand the logic of the code yet so I can't decode what you mean precisely. I will ask some question on MM.

Reply:
Changed it now, is it the way you imagined?

Reply:
Yep, looks good. Thanks!